### PR TITLE
feat: Zoomable image improvements

### PR DESCRIPTION
**Pull request by przemvs**
https://wearezeta.atlassian.net/browse/WPB-1445

## Description

Improved Image zoom. Now user can click on image and zoom image to original width and height of image. User can change image position by dragging on image.

Details copied from JIRA:

### Image gallery view:

- Zoom in is available for any image in the gallery view that is not shown in 100% already
- Cursor is changes to “zoom in” when zoom is available
- When clicking, Image is zoomed to 100% (naturalHeight / naturalWidth)

### Zoomed in view:

- Cursor changes to “zoom out”
- The image zooms in on the part of the image that the mouse pointer pointed to 
- The image can be scrolled in each direction with a smooth scrolling animation
- The image can be dragged in each direction
- No automatic scrolling when the cursor is moved (to reduce fidgety movement on the screen)
- No additional zoom levels (for now) – just zoom in to 100%
- When clicking, image is zoomed back to its previous scale

## Screenshots/Screencast (for UI changes)

https://github.com/wireapp/wire-webapp/assets/13432884/4a4c4c87-823e-425b-827b-2fc502c4bebe

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/ZoomableImage/ZoomableImage.tsx [hunk 1]
```diff
@@ -17,112 +17,230 @@
...
-  return element.naturalWidth !== element.offsetWidth || element.naturalHeight !== element.offsetHeight;
+  const containerRect = containerRef.current.getBoundingClientRect();
+
+  const maxXOffset = (containerRect.width - imgRef.current.naturalWidth) / 2;
+  const maxYOffset = (containerRect.height - imgRef.current.naturalHeight) / 2;
+
+  return {
+    maxXOffset: maxXOffset,
+    maxYOffset: maxYOffset,
+  };
```
Comment (PatrykBuniX) [inline src/script/components/ZoomableImage/ZoomableImage.tsx]: We can shorten it 
```suggestion
  return {
    maxXOffset,
    maxYOffset,
  };
```

File: src/script/components/ZoomableImage/ZoomableImage.tsx [hunk 2]
```diff
@@ -17,112 +17,230 @@
...
 
-  const onButtonClick = (event: React.MouseEvent<HTMLImageElement>) => {
+  const [translateOffset, setTranslateOffset] = useState<Offset>(DEFAULT_OFFSET);
+  const [startOffset, setStartOffset] = useState<Offset>(DEFAULT_OFFSET);
+
+  const canZoomImage = imageRatio !== 1;
+  const zoomScale = isZoomEnabled ? 1 : imageRatio;
+
+  const handleMouseClick = (event: React.MouseEvent<HTMLDivElement>) => {
     const element = event.target as HTMLImageElement;
```
Comment (PatrykBuniX) [inline src/script/components/ZoomableImage/ZoomableImage.tsx]: Hmm, can we actually make sure that the image got clicked? 

Maybe something like:
```js
const element = event.target; // it's any event target here
const image = imageRef.current;

if(!image || element !== image){
  return;
}

// Now we're sure we've clicked on the image, we can type-safely operate on the ref value itself, eg.:
image.style.transition = '';

// [...]
```

Comment (przemvs) [reply src/script/components/ZoomableImage/ZoomableImage.tsx]: Added checking if is Image Element :)

File: src/script/components/ZoomableImage/ZoomableImage.tsx (hunk 1 above)
Comment (przemvs) [reply src/script/components/ZoomableImage/ZoomableImage.tsx]: Done

**Review by PatrykBuniX — APPROVED** [review]
Looks good

**Review by V-Gira — APPROVED** [review]

**Pull request merged**